program/src/duplicate_block_proof.rs

```diff
@@ -144,17 +113,62 @@ impl<'a> SlashingProofData<'a> for DuplicateBlockProofData<'a> {
         if shred2.len() < shred2_length {
             return Err(SlashingError::ProofBufferTooSmall);
         }
+        let (shred2, _) = shred2.split_at(shred2_length);
```

Comment:
Sometimes shreds aren't the full MTU size, so this is actually important since the proof account always stores the full size